GoogleSignIn/Sources/GIDToken.m

```diff
@@ -55,4 +55,37 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
   [encoder encodeObject:_expirationDate forKey:kExpirationDateKey];
 }
 
+#pragma mark - isEqual
+
+- (BOOL)isEqual:(id)object {
+  if (self == object) {
+    return YES;
+  }
+  if (![object isKindOfClass:[GIDToken class]]) {
+    return NO;
+  }
+  return [self isEqualToGIDToken:(GIDToken *)object];
+}
+
+- (BOOL)isEqualToGIDToken:(GIDToken *)other {
+  return [_tokenString isEqual:other.tokenString] &&
+      [self isTheSameDateWithDate1:_expirationDate date2:other.expirationDate];
+}
+
+- (BOOL)isTheSameDateWithDate1:(NSDate *)date1
+                         date2:(NSDate *)date2 {
+  // The date is nullable. Two `null` date is equal.
+  if (!date1 && !date2) {
```

Comment:
Is this the purpose of the method? I'm not sure what the role of this method is given that you call through to `isEqualToDate:`. 

I don't understand why you want nil and nil to return an equal date.

Reply:
This method is to facilitate `isEqualToGIDToken:`. If two tokens have the same token strings and they don't have an expiration date, they are equal.
Do you have any thought to make this clearer? I have not found the best way to compare two nullable properties.

Reply:
I think you should clarify your comment on line 77, which is ambiguous in this context since you don't reference _why_ you're considering two nil dates as equal. A comment like this could clarify the issue: `// Nil dates considered equal so token equality check succeeds if token strings are the same and have no expiration.`

Reply:
That's a good comment!